### PR TITLE
Maze Test and Clean Architecture on Timer

### PR DESCRIPTION
**Pull request by Tomas-Ha**
My original Tests branch was unable to merge due to repackaging. So I downloaded the files, created a new branch or the current origin. Then I re-added my code for the Maze test as well as for how Timer gets its time.

### CODE REVIEW

File: src/tests/entities/MazeTest.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
+package entities;
```
Comment (devan-srinivasan) [inline src/tests/entities/MazeTest.java:1]: Good test suite

File: src/main/interfaceadapters/BuilderSetup.java [hunk 2]
```diff
@@ -7,6 +7,8 @@
 import entities.minigames.*;
```
Comment (devan-srinivasan) [inline src/main/interfaceadapters/BuilderSetup.java:7]: good job adding to `BuilderSetup`

File: src/main/entities/minigames/Maze.java [hunk 3]
```diff
@@ -1,11 +1,11 @@
 package entities.minigames;
 
 import entities.Interactable;
-import entities.minigames.Timer;
 import entities.Unaffordable;
 import entities.interfaces.Moveable;
 import entities.characters.*;
 import entities.weapons.*;
+import usecases.Timing;
```
Comment (devan-srinivasan) [inline src/main/entities/minigames/Maze.java:8]: I was talking with shehzaad and he mentioned Java has a built-in multithread timer. Any reason you chose to make your own?

**Review by devan-srinivasan — APPROVED** [review]

**Comment by devan-srinivasan** [comment]
PR looks good. However, we are waiting to make sure the codebase works for our presentation tomorrow and for submission to the TA. So I may look over it with Shehzaad first because he is overlooking interactable incorporation into Builder, and is making sure everything is ready for submission. @shehzaad777 you good to take a look at this tomorrow? I know its late, sorry. This PR only opened now : /

**Review by devan-srinivasan — COMMENTED** [review]
Hey Tomas. Looks good. Left some comments. Also wanted to ask. You mentioned that you added Clean Architecture on Timer and you previously told me this was in relation to inputting? Maybe I am misremembering (its late lol). Could that be elaborated on?

File: src/main/entities/minigames/Maze.java (hunk 3 above)
Comment (Tomas-Ha) [reply src/main/entities/minigames/Maze.java:8]: It illustrates the CA stuff

**Pull request merged**